### ISSUE TITLE
About haproxy reload and restart command configration

### ISSUE THREAD

**Issue by cylonchau** [haproxytech/dataplaneapi#285]
First of all, I am very grateful to the haproxy developers for their efforts. I have a question with data plane api management haproxy 

## my haproxy.cfg

```
global
  master-worker
  stats socket /var/run/haproxy.sock user haproxy group haproxy mode 660 level admin expose-fd listeners

frontend stats
  mode http
  maxconn 10
  bind *:8404
  stats enable
  stats uri /stats
  stats refresh 10s
  stats hide-version

program api
  command /root/dataplaneapi -f /root/data.hcl
  no option start-on-reload
```
## my dataplaneapi.hcl

```
config_version = 2

name = "normal_walrus"

mode = "single"

dataplaneapi {
  host = "0.0.0.0"
  port = 5555

  user "admin" {
    insecure = true
    password = "111"
  }

  transaction {
    transaction_dir = "/tmp/haproxy"
  }

  advertised {}
}

haproxy {
  config_file = "/root/haproxy.cfg"
  haproxy_bin = "/usr/local/sbin/haproxy"

  reload {
    reload_delay = 5
    reload_cmd   = "kill -SIGUSR2 $(cat /tmp/ha.pid)"
    restart_cmd  = "kill -SIGUSR2 $(cat /tmp/ha.pid)"
  }
}
```
in this config file , my haproxy is not using systemd/initv mode, my haproxy start command is : `haproxy -f /root/haproxy.cfg -p /tmp/ha.pid`.
 
After adding configuration using dataplaneapi, the dataplaneapi use this command to restart or reload haproxy is failure

the error key is: `Reload failed: executing kill -SIGUSR2 $(cat /tmp/ha.pid) failed: exit status 1`

the error log:

```bash
[root@centos-template ~]# haproxy -f /root/haproxy.cfg -p /tmp/ha.pid
[NOTICE]   (9443) : haproxy version is 2.6.7-c55bfdb
[NOTICE]   (9443) : path to executable is /usr/local/sbin/haproxy
[WARNING]  (9443) : config : missing timeouts for frontend 'stats'.
   | While not properly invalid, you will certainly encounter various problems
   | with such a configuration. To fix this, please ensure that all following
   | timeouts are set to a non-zero value: 'client', 'connect', 'server'.
[NOTICE]   (9443) : New program 'api' (9444) forked
[NOTICE]   (9443) : New worker (9445) forked
[NOTICE]   (9443) : Loading success.
time="2022-12-16T22:38:05+08:00" level=warning msg="Restart failed, please check the reason and restart manually: executing kill -SIGUSR2 $(cat /tmp/ha.pid) failed: exit status 1" reload_id=2022-12-16-0
time="2022-12-16T22:38:05+08:00" level=warning msg="Reload failed: executing kill -SIGUSR2 $(cat /tmp/ha.pid) failed: exit status 1" reload_id=2022-12-16-0
```

When using manual execute command ` kill -SIGUSR2 $cat /tmp/ha.pid) ` restart and reload haproxy was successful

dataplaneapi config file field reload_cmd and reload_cmd can not use `kill -SIGUSR2` ?

## environment
- OS: centos7
- execution method: manual execute command `haproxy -f haproxy.cfg -p /tmp/ha.pid -D`
- haproxy version:  2.6.7
- data plane api version:  v2.6.2

**Comment by mjuraga** [comment]
Looking at your configuration, you might have a typo, you are missing `(` here:

   ` reload_cmd   = "kill -SIGUSR2 $cat /tmp/ha.pid)"`

**Comment by cylonchau** [comment]
> Looking at your configuration, you might have a typo, you are missing `(` here:
> 
> ` reload_cmd = "kill -SIGUSR2 $cat /tmp/ha.pid)"`

this configuration was my mistake in copying to here 😅, In actually envirment , the configration is no typo 😢

**Comment by mjuraga** [comment]
While we take a deeper look into this, have you considered running dataplaneapi independently. Running it with `program` section in the haproxy configuration has some caveats and is not really a recommended  way to run HAProxy. Can you confirm that it works that way?

**Comment by shadyabhi** [comment]
@mjuraga I don't want to sidetrack this issue too much, but could you summarize the caveats of running via program directive?

**Comment by cylonchau** [comment]
> While we take a deeper look into this, have you considered running dataplaneapi independently. Running it with `program` section in the haproxy configuration has some caveats and is not really a recommended way to run HAProxy. Can you confirm that it works that way?

Is  ture, if dataplaneapi deployment by binary（not on Docker or Kuberntes, running on linux directly） , due to haproxy or dataplaneapi pid no equie 1(pid 1 is systemd), so the dataplane api and haproxy cannot using `kill -SIGUSR2 $(cat /tmp/ha.pid)` reload. on Docker or Kubernetes work ok. Precent, I migration to docker resovled this reload issue.

**Comment by mjuraga** [comment]
@shadyabhi One of the biggest caveat is that in case anything happens to the dataplaneapi process (unexpected panic or a crash) it would crash your HAProxy process as it's master runner. And we think this is dangerous in a way, as we are sure you don't want your traffic to be down because of an issue with the running sidecar API process.

**Comment by shadyabhi** [comment]
> it would crash your HAProxy process as it's master runner.

@mjuraga Thank you for your response. Can you elaborate on what it means by "master runner"? A link to official docs would be great that explains this problem.
